### PR TITLE
Click tag mentions

### PR DESCRIPTION
**Pull request by poixen**
You can now click on a username in the main chat to add a "@username " tag to the text edit bar. Makes communication with other users using tags faster.

click chat mentions

### CODE REVIEW

**Comment by ZeldaZach** [comment]
Works as expected; good to merge :+1:

**Comment by Daenyth** [comment]
Isn't there a setting to open a private message from clicking username?

File: cockatrice/src/chatview.cpp [hunk 1]
```diff
@@ -249,12 +249,15 @@ void ChatView::mousePressEvent(QMouseEvent *event)
...
         case HoveredUser: {
-            if (event->button() == Qt::RightButton) {
+            if (event->button() != Qt::MidButton) {
                 const int delimiterIndex = hoveredContent.indexOf("_");
-                UserLevelFlags userLevel(hoveredContent.left(delimiterIndex).toInt());
                 const QString userName = hoveredContent.mid(delimiterIndex + 1);
-                
-                userContextMenu->showContextMenu(event->globalPos(), userName, userLevel);
+                if (event->button() == Qt::RightButton) {
+                    UserLevelFlags userLevel(hoveredContent.left(delimiterIndex).toInt());
```
Comment (Daenyth) [inline cockatrice/src/chatview.cpp]: Can you switch on the button instead of chaining if statements?

**Comment by Daenyth** [comment]
I think I might need to test this in relation to #123

**Comment by tooomm** [comment]
Just to make sure...
It's not going to work in the server users list, the players in a game list nor the chat in direct chats, game chats etc.? (you just stated main chat)

**Comment by ZeldaZach** [comment]
I think I'd like it to do what @tooomm suggests (Clicking on the server list would work nice)

**Comment by poixen** [comment]
@tooomm 
Should only be in main chat

@ZeldaZach He was checking where it works.

@Daenyth I cant test that, would be good if you could. I have tried turning on single click and did not get any unexpected results.

**Comment by ZeldaZach** [comment]
is it possible to allow for when single clicking on a name on the userlist for it to input into the chat?

**Comment by poixen** [comment]
@ZeldaZach Doing that would cause unexpected results, when combined with the current double click functionality to open a chat window.

**Comment by tooomm** [comment]
Like to see your functionality in the game chat (same as in main chat, click on username in chat), too. (Targeted) conversations in draft game rooms or multiplayer games is a hassle sometimes.

Additionally add it to the right click menu to make it usable with all appearances of usernames and the related chat in these windows?
e.g. server-user-list or players-in-a-game-list

**Comment by poixen** [comment]
@tooomm added the functionality in-game. I wont be adding the context menu item, as I have other tasks to prioritise. 

@Daenyth hopefully there wont be any issues for you.

I am done with this feature now.

**Comment by tooomm** [comment]
@poixen you are my man :+1:

**Pull request merged**